include/libtorrent/pe_crypto.hpp

```diff
@@ -142,9 +139,8 @@ namespace libtorrent
 		rc4_handler();
 
 		// Input keys must be 20 bytes
-		// TODO: 4 use uint768_t here instead of pointer + length
-		void set_incoming_key(unsigned char const* key, int len) override;
-		void set_outgoing_key(unsigned char const* key, int len) override;
+		void set_incoming_key(span<char const> key) override;
```

Comment:
these keys are fixed length, aren't they? Is uint768_t less convenient than a span? The internal representation is the large uint, right? but maybe it would require conversion and not be very practical for that reason